app/model/block.py

```diff
@@ -10,11 +10,73 @@ def __init__(self):
         self.container = None
         self.questionnaire = None
         self.validation = None
-        self.questionniare = None
+        self.questionnaire = None
         self.templatable_properties = []
         self.display = Display()
 
     def add_section(self, section):
         if section not in self.sections:
             self.sections.append(section)
             section.container = self
+
+    def to_json(self):
+        json_dict = {
+            "id": self.id,
+            "title": self.title,
+            "sections": [],
+            "validation": [],
+            "display": {}
+        }
+
+        for section in self.sections:
+            json_dict['sections'].append(section.to_json())
+
+        if self.validation is not None:
+            for validation in self.validation:
+                json_dict['validation'].append(validation.to_json())
+
+        if self.display is not None:
+            json_dict['display'] = self.display.to_json()
+
+        return json_dict
+
+    def __eq__(self, other):
```

Comment:
As for checking the internal id, the overridden `__eq__` method check that check first as a shortcut before comparing the equivalence of properties (which are also compared for equivalence).

The main issue to be aware of is with collections.  Two equivalent but separate objects may not be stored in a set, for example.  Also, if an object is already in a list, checking that a different but equivalent object is in the list will return true, even through they are different objects